heat/core/operations.py

```diff
@@ -172,8 +172,23 @@ def __local_op(operation, x, out, **kwargs):
         if kwargs_dtype:
             del kwargs['dtype']
         result = operation(x._DNDarray__array.type(torch_type), **kwargs)
-        return dndarray.DNDarray(result, tuple(result.shape), promoted_type if kwargs_dtype is None else kwargs_dtype,
-                                 x.split, x.device, x.comm)
+        split = x.split
+        gshape = list(result.shape)
+        if x.split is not None:
+            # The operation could have reduced the result by one axis
```

Comment:
How is this the case for a local_op? It is just element-wise application of a function. There should never be a reduction happening

Reply:
Some functions (e.g. var()) use the local_op for some local temporary computations (e.g. calculation mean locally and sharing it later). Either we remove this "false" usage or we need to make this case differentiation.